src/huggingface_hub/serialization/_dduf.py

```diff
@@ -0,0 +1,289 @@
+        ...     # Entries will be evaluated lazily when the DDUF file is created (only 1 entry is loaded in memory at a time)
+        ...     yield "vae/config.json", pipe.vae.to_json_string().encode()
+        ...     yield "vae/diffusion_pytorch_model.safetensors", safetensors.torch.save(pipe.vae.state_dict())
+        ...     yield "text_encoder/config.json", pipe.text_encoder.config.to_json_string().encode()
+        ...     yield "text_encoder/model.safetensors", safetensors.torch.save(pipe.text_encoder.state_dict())
+        ...     # ... add more entries here
+
+        >>> export_entries_as_dduf("stable-diffusion-v1-4.dduf", as_entries=as_entries(pipe))
+        ```
+    """
+    logger.info("Exporting DDUF file '%s'", dduf_path)
+    with zipfile.ZipFile(str(dduf_path), "w", zipfile.ZIP_STORED) as archive:
+        for filename, content in entries:
+            if "." + filename.split(".")[-1] not in DDUF_ALLOWED_ENTRIES:
+                raise DDUFExportError(f"File type not allowed: {filename}")
+            if "\\" in filename:
+                raise DDUFExportError(f"Filenames must use UNIX separators: {filename}")
+            logger.debug("Adding file %s to DDUF file", filename)
+            _dump_content_in_archive(archive, filename, content)
+
+    logger.info("Done writing DDUF file %s", dduf_path)
+
+
+def export_folder_as_dduf(dduf_path: Union[str, Path], folder_path: Union[str, Path]) -> None:
+    """
+    Export a folder as a DDUF file.
+
+    AUses [`export_entries_as_dduf`] under the hood.
+
+    Args:
+        dduf_path (`str` or `Path`):
+            The path to the DDUF file to write.
+        folder_path (`str` or `Path`):
+            The path to the folder containing the diffusion model.
+
+    Example:
+        ```python
+        >>> from huggingface_hub import export_folder_as_dduf
+        >>> export_folder_as_dduf("FLUX.1-dev.dduf", diffuser_path="path/to/FLUX.1-dev")
+        ```
+    """
+    folder_path = Path(folder_path)
+
+    def _iterate_over_folder() -> Iterable[Tuple[str, Path]]:
+        for path in Path(folder_path).glob("**/*"):
+            if path.is_dir():
+                continue
+            if path.suffix not in DDUF_ALLOWED_ENTRIES:
+                logger.debug("Skipping file %s (file type not allowed)", path)
+                continue
+            path_in_archive = path.relative_to(folder_path)
+            if len(path_in_archive.parts) > 3:
+                logger.debug("Skipping file %s (nested directories not allowed)", path)
+                continue
+            yield path_in_archive.as_posix(), path
+
+    export_entries_as_dduf(dduf_path, _iterate_over_folder())
+
+
+def add_entry_to_dduf(dduf_path: Union[str, Path], filename: str, content: Union[str, Path, bytes]) -> None:
+    """
+    Add an entry to an existing DDUF file.
```

Comment:
(nit): maybe a note on when this is useful and an example?

Reply:
actually not sure it is useful in the end :smile: I used it during my testing and I do see a possibility where we would like to take an existing DDUF file and add an adapter file to it for instance. That's why I kept it.

I did not document it nor exposed it in `huggingface_hub`